### PR TITLE
976556 - Use hostname for Signo if fqdn is N/A

### PR DESCRIPTION
**Pull request by ares**
addresses https://bugzilla.redhat.com/show_bug.cgi?id=976556

### CODE REVIEW

File: app/models/setting/auth.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def self.load_defaults
     # Check the table exists
     return unless super
 
-    fqdn = Facter.fqdn
+    fqdn = Facter.fqdn || Facter.hostname
```
Comment (ohadlevy) [inline app/models/setting/auth.rb:8]: I'm not sure if this makes sense, we want a properly configured system

**Comment by domcleal** [comment]
We also depend on the domain in a few other settings, so these would need appropriate substitutions.  I can't say I'm very keen on the change, the FQDN is required in the installer and for Puppet to function.

**Comment by ares** [comment]
@ohadlevy @domcleal does this mean that Foreman cannot run on machine without FQDN? This is kind of a hack that is used in katello-configure or foreman-proxy-configure. We don't rely on Facter FQDN guess. See https://bugzilla.redhat.com/show_bug.cgi?id=760265 for more info.

The other fix is to define signo_url during katello-configure which is here https://github.com/Katello/katello-installer/pull/37 and which uses `Socket.gethostname` hack already, however I wanted to configure a meaningful default value.

**Comment by domcleal** [comment]
@ares yes, that's what I'm suggesting.  IIRC, running Puppet without an FQDN is a warning condition and could lead to unpredictable behaviour, particularly if the machine later acquired a domain and so the hostname appears to change.

Regarding the hack, that's improved in Facter 1.7 so it could possibly be removed.

**Comment by ares** [comment]
Ok, closing this one, issue solved by merging https://github.com/Katello/katello-installer/pull/37

**Pull request closed without merging**